instaclustr/resource_cluster.go

```diff
@@ -340,6 +340,24 @@ func resourceCluster() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"private_link": {
+				Type:     schema.TypeList,
```

Comment:
**[QUESTION]** Why `schema.TypeList`?

Reply:
Because TypeMap can't have complex type inside it. That's what Arjun told me last time when I worked on the Terraform ticket.

Reply:
Interesting cause `bundle` is similar where they use `TypeSet` instead. I'll leave it up to the customerPlatform team then :D

Reply:
Oh, I can explain that, because cluster can have several bundles, so Set could be better. But this private_link, we will only have one in the create request, so I gave this to it: `MaxItems: 1`. If max item is 1, there's no difference between List and Set then.